src/js/util/param_spec.js

```diff
@@ -0,0 +1,40 @@
+function ParamSpec(sourceLow, sourceHigh, distLow, distHigh, clamp = true) {
```

Comment:
Is there a better name for this? I don't really know what the `Spec` part conveys?